### PR TITLE
Support LLVM 12.0.0

### PR DESCRIPTION
**Pull request by sherlly**
1. Replace deprecated Callsite/ImmutableCallSite with SVF::CallSite class
2. Fix isNodeHidden() method, which requires two arguments now
3. Fix StringRef to string type conversion
4. Update .travis.yml/build.sh

### CODE REVIEW

**Comment by yuleisui** [comment]
Thanks for this clean pull request! I assume that if we do not change Travis, the patch still works with LLVM-10 for backward compatibility purposes?

**Comment by sherlly** [comment]
> Thanks for this clean pull request! I assume that if we do not change Travis, the patch still works with LLVM-10 for backward compatibility purposes?

Oh, maybe I need to change the isNodeHidden method for backward compatibility purposes.

**Comment by sherlly** [comment]
I already check the changed files carefully, now my LLVM-10 can also work on this version.

File: build.sh [hunk 1]
```diff
@@ -11,13 +11,13 @@
 ########
 SVFHOME=$(pwd)
 sysOS=`uname -s`
-MacLLVM="https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-apple-darwin.tar.xz"
-UbuntuLLVM="https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz"
+MacLLVM="https://github.com/llvm/llvm-project/releases/download/llvmorg-12.0.0-rc3/clang+llvm-12.0.0-rc3-x86_64-apple-darwin.tar.xz"
```
[the remark this answers is not in the document]
Comment (sherlly) [reply build.sh:14]: Yes, the above link is the newest official version about Ubuntu/Mac.

File: .travis.yml [hunk 2]
```diff
@@ -7,14 +7,20 @@ matrix:
...
 env:
   - SVF_CTIR=1 SVF_Z3=1
 
 before_script:
   - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get update       ; fi
-  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install cmake gcc g++ nodejs; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test       ; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get update       ; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo apt-get install cmake gcc-9 g++-9 nodejs; fi
+  - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 900 --slave /usr/bin/g++ g++ /usr/bin/g++-9; fi
```
[the remark this answers is not in the document]
Comment (sherlly) [reply .travis.yml:21]: Using gcc-7 may have linking error (see https://travis-ci.org/github/SVF-tools/SVF/jobs/767457976#L1335), I'll try gcc-8 instead.

[the remark this answers is not in the document]
Comment (sherlly) [reply .travis.yml]: The same linking error occurred in https://travis-ci.org/github/SVF-tools/SVF/jobs/767861847#L1419, I think gcc-9 is a better choice.

[the remark this answers is not in the document]
Comment (yuleisui) [reply .travis.yml:21]: Hmm, it is a bit strange. Previously, it was ok. What is the default version of gcc in this ubuntu?

[the remark this answers is not in the document]
Comment (sherlly) [reply .travis.yml:21]: I remember the default gcc version in CI is 7.5.0 (https://travis-ci.org/github/SVF-tools/SVF/jobs/767457976#L1192). I also found the official GCC mentioned it  (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91067)

[the remark this answers is not in the document]
Comment (yuleisui) [reply .travis.yml:21]: OK. It looks not good... I will check your pull request again tomorrow before we can merge it.

**Pull request merged**